packages/collection/src/collection.ts

```diff
@@ -988,8 +989,20 @@ export class Collection<Key, Value> extends Map<Key, Value> {
 		return [...this.entries()];
 	}
 
+	/**
+	 * Emulates the default sort comparison algorithm used in ECMAScript. Equivalent to calling the
+	 * {@link https://tc39.es/ecma262/multipage/indexed-collections.html#sec-comparearrayelements | CompareArrayElements}
+	 * operation with arguments `firstValue`, `secondValue` and `undefined`.
+	 */
 	private static defaultSort<Value>(firstValue: Value, secondValue: Value): number {
-		return Number(firstValue > secondValue) || Number(firstValue === secondValue) - 1;
+		if (firstValue === undefined) return secondValue === undefined ? 0 : 1;
+		if (secondValue === undefined) return -1;
+
+		const x = String(firstValue);
+		const y = String(secondValue);
```

Comment:
The only difference between the ECMA-262 `ToString` abstract operation and calling the `String()` constructor here, is that the `ToString` operation throws if its argument is a symbol.

I didn't think it was worth adding checks here – if a user is creating a collection of symbols and then trying to sort it, then they probably have bigger issues...

Reply:
I think that's a reasonable compromise, we don't have to follow the specification to a T, and if we added every single thing, the library would probably become a bit big.